src/hotspot/share/gc/g1/g1CollectedHeap.inline.hpp

```diff
@@ -266,15 +267,19 @@ inline void G1CollectedHeap::pin_object(JavaThread* thread, oop obj) {
   assert(obj != nullptr, "obj must not be null");
   assert(!is_gc_active(), "must not pin objects during a GC");
   assert(obj->is_typeArray(), "must be typeArray");
-  HeapRegion *r = heap_region_containing(obj);
-  r->increment_pinned_object_count();
+  HeapRegion* r = heap_region_containing(obj);
+  uint obj_region_idx = r->hrm_index();
```

Comment:
The two can probably be merged to avoid creating `r`.

Reply:
Will do, although the resulting code is the same.